framework/src/main/java/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -825,6 +892,16 @@ public Boolean visitTypevar_Typevar(
       } else if (!subtypeHasAnno && !supertypeHasAnno && areEqualInHierarchy(subtype, supertype)) {
         // two unannotated uses of the same type parameter are of the same type
         return true;
+      } else if (subtypeHasAnno) {
+        Set<AnnotationMirror> superLBs =
+            AnnotatedTypes.findEffectiveLowerBoundAnnotations(qualifierHierarchy, supertype);
+        AnnotationMirror superLB =
+            qualifierHierarchy.findAnnotationInHierarchy(superLBs, currentTop);
+        return qualifierHierarchy.isSubtype(subtype.getAnnotationInHierarchy(currentTop), superLB);
+      } else if (supertypeHasAnno) {
+        return qualifierHierarchy.isSubtype(
+            subtype.getEffectiveAnnotationInHierarchy(currentTop),
+            supertype.getAnnotationInHierarchy(currentTop));
 
       } else if (subtype.getUpperBound().getKind() == TypeKind.INTERSECTION) {
         // This case happens when a type has an intersection bound.  e.g.,
```

Comment:
Three lines below, a comment says "one use of the type has an annotation and the other does not", but I don't think that is possible because the previous two cases handled whenever one use has an annotation and one does not.  Can you double-check?